mgmtsystem_action/reports/mgmtsystem_action_report.py

```diff
@@ -0,0 +1,92 @@
+from odoo import fields, models, tools
+
+
+class MgmtsystemtActionReport(models.Model):
+    """Management System Action Report."""
+
+    _name = "mgmtsystem.action.report"
+    _auto = False
+    _description = "Management System Action Report"
+    _rec_name = "id"
+
+    # Compute data
+    number_of_actions = fields.Integer("# of actions", readonly=True)
+    age = fields.Integer("Age", readonly=True)
+    number_of_days_to_open = fields.Integer("# of days to open", readonly=True)
+    number_of_days_to_close = fields.Integer("# of days to close", readonly=True)
+    number_of_exceedings_days = fields.Integer("# of exceedings days", readonly=True)
+
+    # Grouping view
+    type_action = fields.Selection(
+        [
+            ("immediate", "Immediate Action"),
+            ("correction", "Corrective Action"),
+            ("prevention", "Preventive Action"),
+            ("improvement", "Improvement Opportunity"),
+        ],
+        "Response Type",
+    )
+    create_date = fields.Datetime("Create Date", readonly=True, index=True)
+    date_open = fields.Datetime("Opening Date", readonly=True, index=True)
+    date_closed = fields.Datetime("Close Date", readonly=True, index=True)
+    date_deadline = fields.Date("Deadline", readonly=True, index=True)
+    user_id = fields.Many2one("res.users", "User", readonly=True)
+    stage_id = fields.Many2one("mgmtsystem.action.stage", "Stage", readonly=True)
+    system_id = fields.Many2one("mgmtsystem.system", "System", readonly=True)
+
+    def _query(
```

Comment:
This method is the paradise for sql-injection xD ... Needs a big sign saying "Use under your own responsibility". If other uses it on a public method in a bad way... others could abuse it via rpc.

Reply:
Well, being a private method, the scope is reduced and no external access can exploit, and if you have ORM access, then you may abuse directly without using this method.